src/Entity/TaskListRepository.php

```diff
@@ -0,0 +1,184 @@
+                // https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/reference/dql-doctrine-query-language.html#result-format
+                'item',
+                'tour',
+                'tourItem',
+            ])
+            ->from(TaskList::class, 'tl')
+            ->leftJoin('tl.items', 'item')
+            ->leftJoin('item.tour', 'tour')
+            ->leftJoin('tour.items', 'tourItem')
+            ->where('tl.courier = :courier')
+            ->andWhere('tl.date = :date')
+            ->setParameter('courier', $user)
+            ->setParameter('date', $date)
+            ->getQuery()
+            ->getResult();
+
+        $taskList = $taskListQueryResult[0] ?? null;
+
+        if (null === $taskList) {
+            return null;
+        }
+
+        $orderedTaskIds = array_map(function (Task $task) {
+            return $task->getId();
+        }, $taskList->getTasks());
+
+        $tasksQueryResult = $this->entityManager->createQueryBuilder()
+            ->select([
+                't',
+                'd.id AS deliveryId',
+                'o.number AS orderNumber',
+                'o.total AS orderTotal',
+                'org.name AS organizationName',
+                // objects are listed below to force them being hydrated / pre-fetched by Doctrine
+                // https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/reference/dql-doctrine-query-language.html#result-format
+                'taskPackage',
+                'package',
+                'incidents',
+            ])
+            ->from(Task::class, 't')
+            ->leftJoin('t.delivery', 'd')
+            ->leftJoin('d.order', 'o')
+            ->leftJoin('t.organization', 'org')
+            ->leftJoin('t.packages', 'taskPackage')
+            ->leftJoin('taskPackage.package', 'package')
+            ->leftJoin('t.incidents', 'incidents')
+            ->where('t.id IN (:taskIds)')
+            ->andWhere('t.status != :statusCancelled')
+            ->setParameter('taskIds', $orderedTaskIds) // using IN might cause problems with large number of tasks
+            ->setParameter('statusCancelled', Task::STATUS_CANCELLED)
+            ->getQuery()
+            ->getResult();
+
+        $tasksByDeliveryId = array_reduce($tasksQueryResult, function ($carry, $row) {
+            $deliveryId = $row['deliveryId'] ?? null;
+
+            if (null === $deliveryId) {
+                return $carry;
+            }
+
+            $task = $row[0];
+            $carry[$deliveryId][] = $task;
+            return $carry;
+        }, []);
+
+        $tasks = array_map(function ($row) use ($tasksByDeliveryId) {
+            $task = $row[0];
+            $deliveryId = $row['deliveryId'] ?? null;
+
+            $taskPackages = [];
+            $weight = null;
+
+            $tasksInTheSameDelivery = $deliveryId ? $tasksByDeliveryId[$deliveryId] : [];
+
+            if ($task->isPickup()) {
```

Comment:
cool!
I think i had the reflex to write it in SQL as to go faster but I think it is cool if we move it in a utility function we can test it better and it will act as a spec :+1: